### ISSUE TITLE
Add iterated views

### ISSUE THREAD

**Issue by MattWindsor91** [septract/starling-tool#22]
Starling should allow views to be marked as ‘iterated’:

```
view F()[n];
constraint F()[n] => ...; // some predicate on n
```

meaning that:
- The reification `|_ B * F()[n] _|` is `|_ B _| * F()[n]`, not `|_ B _| * F()[n] * F()[n-1]` etc. (it might be nice to extract this 'closed view' idea and allow it to be applied to views;
- When there is a constraint `P(n)`, we do a quick Z3 check that `n > 0 ^ P(n) => P(n-1)` is satisfiable;
- `n` is treated like a variable arity in multisets, so multiset minus and union work as expected.

**Comment by MattWindsor91** [comment]
Partly fixed by #64 and #65.  When #66 is merged this will be done.

**Comment by MattWindsor91** [comment]
Closed due to everything being merged.
